tests/validator/test_metrics_calculator.py

```diff
@@ -47,13 +47,34 @@ def integer_and_datetime_sample_dataset() -> dict:
     [
         pytest.param("pandas", marks=pytest.mark.unit),
         pytest.param("sqlite", marks=pytest.mark.sqlite),
-        pytest.param("spark", marks=pytest.mark.spark),
     ],
 )
 def test_column_partition_metric(
     sa,
     in_memory_runtime_context,
+    integer_and_datetime_sample_dataset: dict,
+    backend: str,
+):
+    _test_column_partition_metric(
+        sa, in_memory_runtime_context, integer_and_datetime_sample_dataset, backend
+    )
+
+
+@pytest.mark.spark
+def test_column_partition_metric_for_spark(
```

Comment:
I broke out the spark tests from the others because the spark test requires a `spark_session` while the others do not.